src/engine/test/source/api/graph/handlers_test.cpp

```diff
@@ -0,0 +1,185 @@
+#include <api/catalog/catalog.hpp>
+#include <api/graph/handlers.hpp>
+#include <cmds/src/defaultSettings.hpp>
+#include <metrics/metricsManager.hpp>
+#include <mocks/store.hpp>
+#include <unistd.h>
+
+#include <filesystem>
+#include <fstream>
+
+#include <gtest/gtest.h>
+
+#include <testsCommon.hpp>
+
+using namespace api::graph::handlers;
+
+const std::string rCommand {"dummy cmd"};
+const std::string rOrigin {"Dummy org module"};
+constexpr auto ASSET_PATH {"/test/source/api/graph/assets/"};
+constexpr auto BUFFER_PATH_SIZE {1024};
+constexpr auto JSON_DECODER {"decoder/core-hostinfo/0"};
+constexpr auto JSON_FILTER {"filter/allow-all/0"};
+constexpr auto JSON_INTEGRATION {"integration/wazuh-core/0"};
+constexpr auto JSON_POLICY {"policy/wazuh/0"};
+constexpr auto JSON_SCHEMA {"schema/wazuh-logpar-types/0"};
+
+std::string readJsonFile(const std::string& filePath)
+{
+    std::ifstream jsonFile(filePath);
+
+    if (!jsonFile.is_open())
+    {
+        return "";
+    }
+
+    std::stringstream buffer;
+    buffer << jsonFile.rdbuf();
+
+    jsonFile.close();
+
+    return buffer.str();
+}
+
+class MockStoreAdapter : public store::FileDriver
```

Comment:
This adapter class is needed because the graph has the file driver as a dependency instead of the store interface. We should change the dependency to be the interface.